File: Solutions in Java/MaximumDepthOfBinaryTree.java
/**
 * @author : Anmol Agrawal
 */
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int depth = 0;
	void calculateDepth(TreeNode node, int level){
		if (node == null)
			return;
        if(level > depth){
            depth = level;
        }
		calculateDepth(node.left, level + 1);
		calculateDepth(node.right, level + 1);
	}
    public int maxDepth(TreeNode root) {
        calculateDepth(root, 1);
        return depth;
    }
}

File: Solutions in Javascript/MaximumDepthOfBinaryTree.js
/**
 * @author : Anmol Agrawal
 */
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
let depth = 0;
var calculateDepth = function(node, level){
    if (node === null)
        return;
    if(level > depth){
        depth = level;
    }
    calculateDepth(node.left, level + 1);
    calculateDepth(node.right, level + 1);
}
var maxDepth = function(root) {
    depth = 0;
    calculateDepth(root, 1);
    return depth;
};